cocotb/utils.py

```diff
@@ -280,13 +280,13 @@ def sane(x):
         return r
 
     def highlight(string, colour=ANSI.COLOR_HILITE_HEXDIFF_DEFAULT):
-        """Highlight only with ANSI output if it's requested and we are not in a GUI."""
+        """Highlight with ANSI colors if possible/requested and not running in GUI."""
         
-        want_ansi = os.getenv("COCOTB_ANSI_OUTPUT") and not os.getenv("GUI")
-        if want_ansi is None:
-            want_ansi = sys.stdout.isatty()  # default to ANSI for TTYs
-        else:
-            want_ansi = want_ansi == '1'
+        want_ansi = sys.stdout.isatty()  # default to ANSI for TTYs
+        if os.getenv("COCOTB_ANSI_OUTPUT") is not None:
+            want_ansi = True
+        if os.getenv("GUI") is not None:
+            want_ansi = False
```

Comment:
Can these lines be made a helper function to eliminate the duplication?

Reply:
I had the impression it may have been duplicated on purpose, but I'll just add a helper function. Should the name start with an underscore, or is that bad practice when it's known it will be imported in a different module?

Reply:
Would probably skip the underscore, I don't think much in utils.py is named that way